lightning/src/onion_message/messenger.rs

```diff
@@ -800,7 +829,7 @@ where
 			peers.into_iter(),
 			&self.entropy_source,
 			secp_ctx,
-			false,
+			true,
```

Comment:
It's confusing that the docs on the struct say the `NodeIdMessageRouter` always creates non-compact paths, but we set `compact_paths` to `true` here. Can you add a comment?